joy_comms_control_node:
  ros__parameters:
    enable_button: 5
    next_layout_button: "BUTTON_OPTION"
    previous_layout_button: "BUTTON_SHARE"
    controller_type: 2
    #number of commands has to match the number of mappings in controller_mappings and topics in command_topics
    stop_commands:
      ["motors_stop",
       "motors_estop"]
    #number of topics has to match the number of mappings in controller_mappings and commands in stop_commands
    command_topics:
      ["/arm_command",
       "/rover_command"]
    #number of mappings has to match the number of topics in command_topics and commands in stop_commands
    arm_mappings:
      ["TRIGGER_R2,set_motor_speeds %a 0 0 0 0 0,10,250",
      "TRIGGER_L2,set_motor_speeds -%a 0 0 0 0 0,10,250",
      "JOY_LEFT_Y,set_motor_speeds 0 %a 0 0 0 0,10,250",
      "JOY_LEFT_X,set_motor_speeds 0 0 %a 0 0 0,10,250",
      "JOY_RIGHT_Y,set_motor_speeds 0 0 0 %a 0 0,10,250",
      "JOY_RIGHT_X,set_motor_speeds 0 0 0 0 %a 0,10,250",
      "BUTTON_CIRCLE,set_motor_speeds 0 0 0 0 0 250,10,250",
      "BUTTON_SQUARE,set_motor_speeds 0 0 0 0 0 -250,10,250"]
    wheel_mappings:
      ["TRIGGER_L2,move_wheels %a %a %a -%a -%a -%a,10,250",
      "TRIGGER_R2,move_wheels -%a -%a -%a %a %a %a,10,250",
      "JOY_RIGHT_X,move_wheels -%a -%a -%a -%a -%a -%a,10,250",
      "BUTTON_CROSS,blink_toggle 1,10,250",
      "BUTTON_CIRCLE,blink_toggle 0,10,250"]
